Add notice about upcoming service TLS changes to the readme (#2025)
Azure/azure-iot-sdk-c: readme.md

For constained devices - where memory is measured in kilobytes and not megabytes - there are even lighter weight SDK options available.  See [Other Azure IoT SDKs](#other-azure-iot-sdks) for more.

> ## Critical Upcoming Change Notice

Reviewer: Please move this under the table of contents

> All Azure IoT SDK users are advised to be aware of upcoming certificate changes for IoT hub and Device Provisioning Service 

Reviewer: ```suggestion
All Azure IoT SDK users are advised to be aware of upcoming TLS certificate changes for Azure IoT Hub and Device Provisioning Service 
```

> that will impact the SDK's ability to connect. In October 2022, both services will migrate from the current 

Reviewer: ```suggestion
that will impact the SDK's ability to connect to these services. In October 2022, both services will migrate from the current 
```

[Baltimore CyberTrust CA Root](https://baltimore-cybertrust-root.chain-demos.digicert.com/info/index.html) to the 
[DigiCert Global G2 CA root](https://global-root-g2.chain-demos.digicert.com/info/index.html). There will be a 
transition period beforehand where your IoT devices must have both the Baltimore and Digicert public certificates 
installed in their certificate store in order to prevent connectivity issues. 

> **Devices with only the Baltimore public certificate installed will lose the ability to connect to IoT hub and Device Provisioning Service in October 2022.**

Reviewer: ```suggestion
**Devices with only the Baltimore public certificate installed will lose the ability to connect to Azure IoT Hub and Device Provisioning Service in October 2022.**
```

> To prepare for this change, make sure your device's certificate store has both of these public certificates installed.

Reviewer: ```suggestion
To prepare for this change, make sure your device's TLS stack has both of these public certificates configured.
```

> For a more in depth explanation as to why the service is doing this, please see

Reviewer: ```suggestion
For a more in depth explanation as to why the IoT service is doing this, please see
```

  - [Schedule<sup>1</sup>](#schedulesup1sup)
    - [Planned Release Schedule](#planned-release-schedule)

> ## Critical Upcoming Change Notice

Reviewer: Please ass this to the TOC above.

All Azure IoT SDK users are advised to be aware of upcoming TLS certificate changes for Azure IoT Hub and Device Provisioning Service 
that will impact the SDK's ability to connect to these services. In October 2022, both services will migrate from the current 
[Baltimore CyberTrust CA Root](https://baltimore-cybertrust-root.chain-demos.digicert.com/info/index.html) to the 
[DigiCert Global G2 CA root](https://global-root-g2.chain-demos.digicert.com/info/index.html). There will be a 
transition period beforehand where your IoT devices must have both the Baltimore and Digicert public certificates 

> hardcoded in their application or flashed onto your WiFi module in order to prevent connectivity issues. 

Reviewer: ```suggestion
which maybe hardcoded in their application or flashed onto your WiFi module in order to prevent connectivity issues. 
```

Author: "which may be...", but yeah I like this change

**Devices with only the Baltimore public certificate will lose the ability to connect to Azure IoT Hub and Device Provisioning Service in October 2022.**

> To prepare for this change, make sure your device's TLS stack has both of these public certificates configured.

Reviewer: ```suggestion
To prepare for this change, make sure your device's TLS stack has both of these public root of trust certificates configured.
```
